modAionImpl/src/org/aion/zero/impl/db/TransactionStore.java

```diff
@@ -50,23 +56,56 @@ public boolean putToBatch(AionTxInfo tx) {
                 }
             }
             existingInfos.add(tx);
-            source.putToBatch(txHash, existingInfos);
+            txInfoSource.putToBatch(txHash, existingInfos);
 
             return true;
         } finally {
             lock.writeLock().unlock();
         }
     }
 
-    public void flushBatch() {
-        source.flushBatch();
+    public void putAliasesToBatch(AionTxInfo txInfo) {
```

Comment:
since this functionality requires the `txInfo` I would prefer if it were done as part of the `putToBatch`/`putTxInfoToBatch`

Reply:
That would have been ideal but most of the time the txInfo passed into putTxInfoToBatch does not include the internal transactions